app/classes/query/scope_modules/initialization.rb

```diff
@@ -0,0 +1,223 @@
+# frozen_string_literal: true
+
+# Helper methods for turning Query parameters into AR conditions.
+module Query::ScopeModules::Initialization
+  attr_accessor :scopes, :last_query
+
+  def initialized?
+    @initialized ? true : false
+  end
+
+  def initialize_query
+    @initialized = true
+    @scopes      = model
+    initialize_scopes
+    @last_query = sql
+  end
+
+  def sql
+    initialize_query unless initialized?
+
+    @sql = scopes.all.to_sql
+  end
+
+  def query
+    initialize_query unless initialized?
+
+    @query = scopes.all
+  end
+
+  def initialize_scopes
+    initialize_parameter_set
+    filter_misspellings_for_name_queries
+    send_rss_log_content_filters_to_subqueries
+    add_default_order_if_none_specified
+  end
+
+  # For transition only
+  def initialize_non_nil_defaults; end
+
+  def initialize_parameter_set
+    sendable_params.each do |param, val|
+      next if (param != :id_in_set && skippable_values.include?(val.to_s)) ||
+              (param == :id_in_set && val.nil?) # keep empty array
+
+      @scopes = if val.is_a?(Hash)
+                  @scopes.send(param, **val)
+                else
+                  @scopes.send(param, val)
+                end
+    end
+  end
+
+  # We don't `compact` sendable_params, in order to keep empty arrays for
+  # `:id_in_set`. We also do want `false` values, so we can't check `blank?`
+  def skippable_values
+    @skippable_values = ["[]", "{}", "", nil].freeze
+  end
+
+  # For RssLogs, remove any content filter params before passing to scopes
+  # since they're already handled in subqueries above.
+  # Otherwise, these are the `scope_parameters` defined in Query::Base.
+  def sendable_params
+    sendable = params.slice(*scope_parameters)
+    return sendable unless model == RssLog
+
+    sendable.except(*content_filter_parameters.keys)
+  end
+
+  # Most name queries are filtered to remove misspellings.
+  def filter_misspellings_for_name_queries
+    return if model != Name || !params[:misspellings].nil?
+
+    @scopes = @scopes.with_correct_spelling
+  end
+
+  # In the case of RssLogs, send any content filter params to subqueries.
+  # (Content filters may add params to RssLog queries that RssLog scopes
+  # can't handle, because they're intended for one or more related models.)
+  # Some params may go into more than one subquery if >1 `type` requested.
+  def send_rss_log_content_filters_to_subqueries
+    return if model != RssLog || !content_filters_present
+
+    rss_logs_requested_types.each do |model|
+      subquery_params = content_filter_subquery_params(model)
+      if subquery_params.present?
+        @scopes = @scopes.send(:"#{model.name.downcase}_query",
+                               **subquery_params)
+      end
+    end
+  end
+
+  # Current types requested on the RssLog page. Defaults to :all.
+  def rss_logs_requested_types
+    types = [:observation, :name, :location]
```

Comment:
Do you also need Species Lists, Projects, Glossary, News?
<img width="903" alt="Screenshot 2025-03-31 at 7 04 12 AM" src="https://github.com/user-attachments/assets/f9810891-a252-48cd-b7a8-19c34c6da46f" />

Reply:
Added